python/simulation.cpp

```diff
@@ -232,6 +249,13 @@ void register_simulation(pybind11::module& m, pyarb_global_ptr global_ptr) {
              "Rebuild the connection table from recipe::connections_on and the event"
              "generators based on recipe::event_generators.",
              "recipe"_a)
+        .def("deserialize", &simulation_shim::deserialize,
```

Comment:
Cute indeed, but no clue if it'll work, since this is currently legal (and needed)
```cxx
std::unique_ptr<double> pd = std::make_unique(42.0);
serialize(sd, pd);
*pd = 23;
deserialize(sd, pd);
```

Reply:
But that's not more of an issue when we call it 'Pickle', is it?

(Could even pass the json to ZipFile)

Reply:
Pickle is a known, well-defined term with its own protocol. I don't think we can overload that name.

Reply:
From the linked docs, I understood this is true for Python objects, but not for C++-Pybind ones, and that that's the reason you can implement these methods at will.

Looking at the [Python pickle docs](https://docs.python.org/3/library/pickle.html), the warning is clear.

Reply:
I am not sure which warning you're referring to, but Pickle defines a protocol ie
https://docs.python.org/3/library/pickle.html#data-stream-format
that we do not adhere to. My comment about pointers was referring to the fact
that this PR relies the structure being in place already, so a more accurate counter
example would be this
```cxx
int* pi = nulltpr;
deserialize(sd, pid); // BOOM, tries to deref pi and assign to its pointee
pi = new int;
deserialize(sd, pid); // Fine, we now have a value to write to.
```